plugin/al/translators/MayaReference.cpp

```diff
@@ -213,23 +226,43 @@ MStatus MayaReferenceLogic::update(const UsdPrim& prim, MObject parent) const
     }
   }
 
-  // Check to see if we already have a reference matching the prim's namespace.
+  // Check to see whether we have previously created a reference node for this
+  // prim.  If so, we can just reuse it.
+  //
+  // The check is based on comparing the prim's full path and the name of the
+  // reference node, which we had originally created from the prim's full
+  // path.  (Because of this, if the name or parentage of the prim has changed
+  // since we created the reference, we won't find the old reference node.
+  // The old one will be left orphaned, a new one will be created, and we will
+  // lose any reference edits we had made.  Ideally, this won't happen often.
+  // To prevent the problem, we could store the name of the reference node in
+  // an attribute on the prim node.  The reference node would never be renamed
+  // by the user, since it's locked.  If the user were to rename the prim
+  // node, we could update the reference node's name too, for consistency,
+  // when the two nodes got reattached.  Not doing this currently, but can
+  // revisit if renaming/reparenting of prims with reference nodes turns out
+  // to be a common thing in the workflow.)
+  //
   if(refNode.isNull())
   {
     for (MItDependencyNodes refIter(MFn::kReference); !refIter.isDone(); refIter.next()) {
       MObject tempRefNode = refIter.item();
       MFnReference tempRefFn(tempRefNode);
       if (!tempRefFn.isFromReferencedFile()) {
-        MPlug primNSPlug = tempRefFn.findPlug(MString(m_primNSAttr), true, &status);
-        if (status == MS::kInvalidParameter) {
-          // No prim NS attribute. These aren't the droids we're looking for.
-          continue;
-        }
 
-        // Test the namespace
-        if (primNSPlug.asString() == rigNamespaceM) {
-          // We found one with the same namespace - continue running an update
+        // Get the name of the reference node so we can check if this node
+        // matches the prim we are processing.  Strip off the "_RN" suffix.
+        //
+        MString refNodeName = tempRefFn.name();
+        MString refName = refNodeName.substring(0, refNodeName.numChars()-4);
```

Comment:
Rather than creating a temp (and the mem allocation that involves), just initialise the string directly. 
```c++
MString refName(refNodeName.asChar(), refNodeName.numChars() - 4);
```

Reply:
Fixed.